kong/db/strategies/postgres/init.lua

```diff
@@ -483,22 +479,62 @@ local function execute(strategy, statement_name, attributes, is_update)
 
   clear_tab(argv)
 
-  for i = 1, argc do
-    local name  = argn[i]
-    local value
+  local is_update = options and options.update
+  local has_ttl = strategy.schema.ttl
+  local ttl_value
+
+  if has_ttl then
+    ttl_value = options and options.ttl
+    if ttl_value then
+      if ttl_value == 0 then
+        ttl_value = escape_literal(connector, null, fields.ttl)
+
+      elseif not is_update and
+             attributes.created_at and
+             fields.created_at and
+             fields.created_at.timestamp and
+             fields.created_at.auto then
+        ttl_value = escape_literal(connector, ttl_value + attributes.created_at, fields.ttl)
+
+      elseif is_update and
+             attributes.updated_at and
+             fields.updated_at and
+             fields.updated_at.timestamp and
+             fields.updated_at.auto then
+        ttl_value = escape_literal(connector, ttl_value + attributes.updated_at, fields.ttl)
 
-    if i == argc and is_update and attributes[UNIQUE] then
-      value = attributes[UNIQUE]
+      else
+        ttl_value = escape_literal(connector, ttl_value + time(), fields.ttl)
```

Comment:
It seems to me like it would be fine to just have this branch? The only purpose the above branches seem to serve is to not repeat the `ngx.time()` call, and use `created_at`, or `updated_at` (which were generated by `ngx.time()` since they have the `auto` attribute). That would simplify the implementation quite a bit, unless those branches serve another purpose that I missed, of course.